src/broker/broker.c

```diff
@@ -661,9 +662,63 @@ static void set_proctitle (uint32_t rank)
     (void)prctl (PR_SET_NAME, proctitle, 0, 0, 0);
 }
 
+int etc_shells_to_argz (char **argz, size_t *argz_len)
+{
+    void *buf = NULL;
+    ssize_t len;
+    int fd = -1;;
+    int rc = -1;
+    int saved_errno;
+
+    if ((fd = open ("/etc/shells", O_RDONLY)) < 0
+        || (len = read_all (fd, &buf)) < 0
+        || (argz_create_sep (buf, '\n', argz, argz_len)) != 0)
```

Comment:
on my box there is a comment in `/etc/shells`, which i guess just gets slurped up here and compared below?  No big deal, but maybe want to mention as a comment?

Reply:
Oh good catch, I should have noticed that. It should be easy to skip lines that start with `#`

Reply:
Hm, I just discovered [`getusershell(3)`](https://linux.die.net/man/3/getusershell), maybe I should just use that here.

Reply:
Ok, pushed a new version using `getusershell(3)` which simplifies this code.